zerver/tests/test_auth_backends.py

```diff
@@ -1472,6 +1479,30 @@ def test_social_auth_invited_as_admin_but_expired(self) -> None:
         created_user = get_user_by_delivery_email(email, realm)
         self.assertEqual(created_user.role, UserProfile.ROLE_MEMBER)
 
+    def do_email_change(self, user_profile: UserProfile, action_key: str,
+                        different_account: bool=False) -> HttpResponse:
+        if different_account:
+            user_profile = self.example_user('cordelia')
```

Comment:
```suggestion
        if different_account:
            assert user_profile != self.example_user('cordelia')
            user_profile = self.example_user('cordelia')
```

to avoid weird bugs if somebody passes cordelia as the user_profile argument

Reply:
Didn't think about that, interesting, Thanks! Added the suggestion.